src/assignments/yunfeng/lesson_4_answers.clj

```diff
@@ -0,0 +1,138 @@
+;;recogize a card problem
+(def suit-map {:S :spade :H :heart :D :diamond :C :club})
+(def rank-map {:2 0 :3 1 :4 2 :5 3 :6 4 :7 5 :8 6 :9 7 :T 8 :J 9 :Q 10 :K 11 :A 12})
+
+(defn parse-suit
+  "parse the string to get the suit and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 0 1)))
+
+;;test parse-suit function
+(parse-suit "ST")
+
+(defn parse-rank
+  "parse the string to get the rank and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 1)))
+
+;;test parse-rank function
+(parse-rank "ST")
+
+
+(defn recogize-a-card
+  "use keywords to find its correspond values and construct the result"
+  [card-string]
+  {:suit ((parse-suit card-string) suit-map) :rank ((parse-rank card-string) rank-map)})
+
+;;test recongize-a-card function
+(recogize-a-card "ST")
+(recogize-a-card "SA")
+(recogize-a-card "H4")
+
+
+
+;;best hand problem
+(defn remove-duplicate-ranks
+  "It is trying to remove duplicate ranks in a hand regardless of what the suit is and return the counted result"
+  [recongized-hand]
+  (->> (map :rank recongized-hand)
+       (set)
+       (count)))
+
+;;test remove-duplicate-ranks function
+(remove-duplicate-ranks [{:suit :spade :rank 12} {:suit :club :rank 12} {:suit :heart :rank 12} {:suit :diamond :rank 12} {:suit :spade :rank 3}])
```

Comment:
Perhaps writing these forms using deftest, in a test file, would be clearer as to what this function is meant to return?

Reply:
Sure I will create a test file